share/system/cgroup_linux.go

```diff
@@ -332,76 +332,147 @@ func getCgroupPath_cgroup_v2(pid int) (string, error) {
 
 }
 
+// GetContainerCgroupPath - Gets a PID's cgroup path for a subsystem
+// PIDs can share containers so we can derive the pid's cgroup path by parsing proc/<pid>/cgroup
 func (s *SystemTools) GetContainerCgroupPath(pid int, subsystem string) (string, error) {
-	/*
-		mnt, err := FindCgroupMountpoint(subsystem)
-		if err != nil {
-			return "", nil
-		}
-	*/
+	subsystemPath := ""
+	mpath, err  := s.getCgroupMetricsPath(pid, subsystem)
+	if err != nil {
+		// Just return the fallback, callers don't care about the error!
+		return mpath, nil
+	}
 
-	// Alternative: it might not be necessary to obtain cgroup path as before
-	var path string
+	// Join to the system's path for host's cgroup
+	subsystemPath = s.JoinToCgroupPath(mpath, subsystem)
```

Comment:
This is the biggest difference in logic. 

Previously, we assumed /proc/<pid>/root/sys/fs/cgroup will always exists. 

The new logic does away with this assumption and instead follows the convention that the container runtime will store the cgroup data in /sys/fs/cgroup (on the host, in the container we expect this to be mounted to /host/cgroup).